homeassistant/components/aurora_abb_powerone/config_flow.py

```diff
@@ -0,0 +1,148 @@
+
+# pylint doesn't think DOMAIN is used.  But it is used!
+# Might be this bug? https://github.com/PyCQA/pylint/issues/3445
+# pylint: disable=unused-import
+from .const import (
+    ATTR_FIRMWARE,
+    ATTR_MODEL,
+    ATTR_SERIAL_NUMBER,
+    CONF_USEDUMMYONFAIL,
+    DEBUGMODE,
+    DEFAULT_ADDRESS,
+    DEFAULT_INTEGRATION_TITLE,
+    DOMAIN,
+    MAX_ADDRESS,
+    MIN_ADDRESS,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def validate_and_connect(hass: core.HomeAssistant, data, populate_on_fail: False):
+    """Validate the user input allows us to connect.
+
+    Data has the keys from DATA_SCHEMA with values provided by the user.
+    """
+    comport = data[CONF_PORT]
+    address = data[CONF_ADDRESS]
+    _LOGGER.debug("Intitialising com port=%s", comport)
+    ret = {}
+    ret["title"] = DEFAULT_INTEGRATION_TITLE
+    try:
+        client = AuroraSerialClient(address, comport, parity="N", timeout=1)
+        client.connect()
+        ret[ATTR_SERIAL_NUMBER] = client.serial_number()
+        ver = client.version()
+        partnum = client.pn()
+        ret[ATTR_MODEL] = f"{ver} ({partnum})"
+        ret[ATTR_FIRMWARE] = client.firmware(1)
+        _LOGGER.info("Returning device info=%s", ret)
+    except AuroraError as err:
+        _LOGGER.warning("Could not connect to device=%s", comport)
+        if populate_on_fail:
+            ret = {
+                "title": DEFAULT_INTEGRATION_TITLE,
+                "serial_number": "735492",
+                "pn": "-3G97-",
+                "firmware": "C.0.3.5",
+            }
+            _LOGGER.warning("Using dummy device info=%s", ret)
+        else:
+            raise err
+    finally:
+        if client.serline.isOpen():
+            client.close()
+
+    # Return info we want to store in the config entry.
+    return ret
+
+
+class AuroraABBConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Aurora ABB PowerOne."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+    _comportslist = None
+    _defaultcomport = None
+
+    def __init__(self):
+        """Initialise the config flow."""
+        self.config = None
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initialised by the user."""
+
+        errors = {}
+        if self._comportslist is None:
+            comports = serial.tools.list_ports.comports(include_links=True)
+            comportslist = []
+            for port in comports:
+                comportslist.append(port.device)
+                _LOGGER.debug("COM port option: %s", port.device)
+            if len(comportslist) > 0:
+                defaultcomport = comportslist[0]
+            else:
+                _LOGGER.warning(
+                    "No com ports found.  Need a valid RS485 device to communicate."
+                )
+                defaultcomport = "No COM ports found"
+                errors["base"] = "no_serial_ports"
```

Comment:
This is not a base error. This should be an abort as it's not recoverable by entering different data. `return self.async_abort(reason='no_serial_ports')`

Reply:
Done.